src/autoschedulers/mullapudi2016/AutoSchedule.cpp

```diff
@@ -2789,25 +3221,37 @@ void Partitioner::generate_group_cpu_schedule(
             if ((iter != stg_estimates.end()) && iter->second.defined()) {
                 if (!seq_var.empty()) {
                     VarOrRVar seq(seq_var, (rvars.find(seq_var) != rvars.end()));
-                    f_handle.reorder(seq, v);
+                    if (t.has_gpu_feature()) {
+                        gpu_tiling.canReorder({seq, v});
+                    } else {
+                        f_handle.reorder(seq, v);
+                        sched.push_schedule(f_handle.name(), g.output.stage_num,
+                                            "reorder(" + seq_var + ", " + var + ")",
+                                            {seq_var, var});
+                    }
+                }
+                if (t.has_gpu_feature()) {
+                    gpu_tiling.canParallelize(v, iter->second);
+                } else {
+                    f_handle.parallel(v);
                     sched.push_schedule(f_handle.name(), g.output.stage_num,
-                                        "reorder(" + seq_var + ", " + var + ")",
-                                        {seq_var, var});
+                                        "parallel(" + var + ")", {var});
                 }
-                f_handle.parallel(v);
-                sched.push_schedule(f_handle.name(), g.output.stage_num,
-                                    "parallel(" + var + ")", {var});
                 def_par = simplify(def_par * iter->second);
             } else {
                 break;
             }
         }
     }
 
-    if (can_prove(def_par < arch_params.parallelism)) {
+    if (can_prove(def_par < arch_params.parallelism && def_par != 1)) {
         user_warning << "Insufficient parallelism for " << f_handle.name() << "\n";
     }
```

Comment:
Question for reviewers: why are we permitting `parallel(y)` having `Func.bound(y, 1)`, i.e. having an extent of one?

See also: #7696 .